File: web/src/app/service/equipment.service.ts
import { Injectable } from '@angular/core';
import {Observable} from 'rxjs';
import {Page} from '../base/page';
import {User} from '../func/User';
import {CommonService} from './common.service';
import {HttpClient} from '@angular/common/http';
import {Router} from '@angular/router';
import {Equipment} from '../func/Equipment';

@Injectable({
  providedIn: 'root'
})
export class EquipmentService {
  private url = 'equipment';
  constructor(private commonService: CommonService,
              private httpClient: HttpClient,
              private router: Router) { }

  /**
   * 分页方法
   * @param page 第几页
   * @param size 每页条数
   * @param userId 用户
   */
  public getToRepair(page: number, size: number): Observable<Page<Equipment>> {
    const params: { [key: string]: any } = {
      page: String(page),
      size: String(size),
    };

    return this.httpClient.get<Page<Equipment>>(`${this.url}/getToRepair`, {params});
  }

  save(equipment: Equipment): any {
    console.log(equipment);
    return this.httpClient.post(`${this.url}`, equipment);
  }

  public delete(equipmentId: number): Observable<null> {
    return this.httpClient.delete<null>(`${this.url}/${equipmentId.toString()}`);
  }

  /**
   * 更新
   */
  public report(equipmentId: number, equipment: Equipment): Observable<Equipment> {
    return this.httpClient.put<Equipment>(`${this.url}/report/${equipmentId.toString()}`, equipment);
  }

  /**
   * 更新
   */
  public repair(equipmentId: number, equipment: Equipment): Observable<Equipment> {
    return this.httpClient.put<Equipment>(`${this.url}/repair/${equipmentId.toString()}`, equipment);
  }

  public getAll(page: number, size: number): Observable<Page<Equipment>> {
    const params: { [key: string]: any } = {
      page: String(page),
      size: String(size),
    };

    return this.httpClient.get<Page<Equipment>>(`${this.url}/getAll`, {params});
  }

  /**
   * 更新
   */
  public scrap(equipmentId: number, equipment: Equipment): Observable<Equipment> {
    return this.httpClient.put<Equipment>(`${this.url}/scrap/${equipmentId.toString()}`, equipment);
  }

  /**
   * 根据id获取订单
   * @param id  订单id
   */
  public getById(id: number): Observable<Equipment> {
    return this.httpClient.get<Equipment>(`${this.url}/${id}`);
  }

  /**
   * 更新
   */
  update(equipmentId: number, equipment: Equipment): Observable<Equipment> {
    return this.httpClient.put<Equipment>(`${this.url}/${equipmentId.toString()}`, equipment);
  }
}

File: api/src/main/java/equipmentManagementSystem/service/EquipmentService.java
package equipmentManagementSystem.service;

import equipmentManagementSystem.entity.Department;
import equipmentManagementSystem.entity.Equipment;
import equipmentManagementSystem.entity.Type;
import equipmentManagementSystem.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;

public interface EquipmentService {
    Page<Equipment> findAll(Pageable pageable);

    Page<Equipment> getToRepair(Pageable pageable);

    void add(Equipment equipment);

    /**
     * 删除
     *
     * @param id
     */
    void delete(Long id);

    Equipment report(Long id, Equipment equipment);

    Equipment repair(Long id, Equipment equipment);


    Equipment scrap(Long id, Equipment equipment);

    Equipment update(Long id, Equipment equipment);

    Equipment getEquipmentById(Long id);

    Page<Equipment> quaryAll(String name, Long states, String place, String internalNumber, Pageable pageable, Long type);

    Equipment borrow(Long id, Equipment equipment);

    Equipment toReturn(Long id, Equipment equipment);

    Page<Equipment> getBorrow(Pageable pageable);
}
